src/seedu/addressbook/data/person/Address.java

```diff
@@ -37,23 +34,9 @@ public static boolean isValidAddress(String test) {
     }
 
     @Override
-    public String toString() {
-        return value;
-    }
-
-    @Override
     public boolean equals(Object other) {
         return other == this // short circuit if same object
                 || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
-    }
-
-    @Override
-    public int hashCode() {
-        return value.hashCode();
-    }
-
-    public boolean isPrivate() {
-        return isPrivate;
+                && this.toString().equals(((Address) other).toString())); // state check
```

Comment:
Since you declare `value` as `protected` then the child class can just call `.value`.